Library/src/Systems/Assets/AssetSystem.cpp

```diff
@@ -847,4 +881,305 @@ CSP_ASYNC_RESULT_WITH_PROGRESS void
+	};
+
+	const csp::common::String MaterialCollectionName = CreateUniqueMaterialAssetCollectionName(Name, SpaceId);
+
+	CreateAssetCollection(SpaceId, nullptr, MaterialCollectionName, nullptr, EAssetCollectionType::DEFAULT, nullptr, CreateAssetCollectionCB);
+}
+
+void AssetSystem::UpdateMaterial(const GLTFMaterial& Material, NullResultCallback Callback)
+{
+	// 1. Get asset collection
+	auto GetAssetCollectionCB = [this, Material, Callback](const AssetCollectionResult& CreateAssetCollectionResult)
+	{
+		if (CreateAssetCollectionResult.GetResultCode() != EResultCode::Success)
+		{
+			Callback(NullResult(CreateAssetCollectionResult.GetResultCode(), CreateAssetCollectionResult.GetHttpResultCode()));
+			return;
+		}
+
+		// 2. Get asset
+		const AssetCollection& CreatedAssetCollection = CreateAssetCollectionResult.GetAssetCollection();
+
+		auto GetAssetCB = [this, Callback, Material, CreatedAssetCollection](const AssetResult& CreateAssetResult)
+		{
+			if (CreateAssetResult.GetResultCode() != EResultCode::Success)
+			{
+				Callback(NullResult(CreateAssetResult.GetResultCode(), CreateAssetResult.GetHttpResultCode()));
+				return;
+			}
+
+			// 3. Upload material
+			auto UploadMaterialCallback = [this, Callback, Material](const UriResult& UploadResult)
+			{
+				Callback(NullResult(UploadResult.GetResultCode(), UploadResult.GetHttpResultCode()));
+			};
+
+			csp::common::String MaterialJson = json::JsonSerializer::Serialize(Material);
+			const Asset& CreatedAsset		 = CreateAssetResult.GetAsset();
+
+			// Create a new string to prevent const casting
+			std::string Buffer(MaterialJson.c_str());
+
+			BufferAssetDataSource AssetData;
+			AssetData.SetMimeType("application/json");
+			AssetData.Buffer	   = Buffer.data();
+			AssetData.BufferLength = MaterialJson.Length();
+
+			UploadAssetData(CreatedAssetCollection, CreatedAsset, AssetData, UploadMaterialCallback);
+		};
+
+		GetAssetById(Material.GetAssetCollectionId(), Material.GetAssetId(), GetAssetCB);
+	};
+
+	GetAssetCollectionById(Material.GetAssetCollectionId(), GetAssetCollectionCB);
+}
+
+void AssetSystem::DeleteMaterial(const GLTFMaterial& Material, NullResultCallback Callback)
+{
+	// 1. Delete asset
+	auto DeleteAssetCB = [this, Callback, &Material](const NullResult& DeleteAssetResult)
+	{
+		if (DeleteAssetResult.GetResultCode() != EResultCode::Success)
+		{
+			Callback(NullResult(DeleteAssetResult.GetResultCode(), DeleteAssetResult.GetHttpResultCode()));
+			return;
+		}
+
+		// 2. Delete asset collection
+		auto DeleteAssetCollectionCB = [this, Callback](const NullResult& DeleteAssetCollectionResult)
+		{
+			Callback(DeleteAssetCollectionResult);
+		};
+
+		DeleteAssetCollectionById(Material.GetAssetCollectionId(), DeleteAssetCollectionCB);
+	};
+
+	DeleteAssetById(Material.GetAssetCollectionId(), Material.GetAssetId(), DeleteAssetCB);
```

Comment:
If the associated AssetCollection only had this one material asset does that get cleaned up by CHS? If not do we need to clear it up?